macrosynergy/download/dataquery.py

```diff
@@ -0,0 +1,743 @@
+                    f"dq_api.Interface.last_url : {self.last_url}"
+                )
+            else:
+                logger.info(f"Request successful.")
+                if select in response.keys():
+                    results.extend(response[select])
+
+                if response["links"][1]["next"] is None:
+                    break
+
+                url = f"{self.access.base_url:s}{response['links'][1]['next']:s}"
+                params = {}
+
+        if (counter > server_count) or (invalid_responses > server_count):
+            raise ConnectionError(
+                f"Connection to DataQuery failed. counter: {counter}, invalid_responses: {invalid_responses}"
+                f"dq_api.Interface.last_url : {self.last_url},"
+                f"dq_api.Interface.last_response : {self.last_response},"
+            )
+
+        if (len(results) == 0) or (None in results):
+            return None
+        else:
+            return results, status, msg
+
+    def _request(
+        self,
+        endpoint: str,
+        tickers: List[str],
+        params: dict,
+        delay: int = 0,
+        count: int = 0,
+        start_date: str = None,
+        end_date: str = None,
+        calendar: str = "CAL_ALLDAYS",
+        frequency: str = "FREQ_DAY",
+        conversion: str = "CONV_LASTBUS_ABS",
+        nan_treatment: str = "NA_NOTHING",
+        debug: bool = False,
+    ):
+        """
+        Method designed to concurrently request tickers from the API. Each initiated
+        thread will handle batches of 20 tickers, and 10 threads will be active
+        concurrently. Able to request data sequentially if required or server overload.
+
+        :param <str> endpoint: url.
+        :param <List[str]> tickers: List of Tickers.
+        :param <dict> params: dictionary of required parameters for request.
+        :param <Integer> delay: each release of a thread requires a delay (roughly 200
+            milliseconds) to prevent overwhelming DataQuery. Computed dynamically if DQ
+            is being hit too hard. Naturally, if the code is run sequentially, the delay
+            parameter is not applicable. Thus, default value is zero.
+        :param <Integer> count: tracks the number of recursive calls of the method. The
+            first call requires defining the parameter dictionary used for the request
+            API.
+        :param <str> start_date:
+        :param <str> end_date:
+        :param <str> calendar:
+        :param <str> frequency: frequency metric - default is daily.
+        :param <str> conversion:
+        :param <str> nan_treatment:
+
+        return <dict>: single dictionary containing all the requested Tickers and their
+            respective time-series over the defined dates.
+        """
+
+        if delay > 0.9999:
+            error_delay = "Issue with DataQuery - requests should not be throttled."
+            raise RuntimeError(error_delay)
+
+        no_tickers = len(tickers)
```

Comment:
We should rename tickers to `expressions` to follow the DataQuery convention in this part.